tests/KubernetesClient.Tests/KubernetesJsonTests.cs

```diff
@@ -41,8 +42,33 @@ public void RFC3339()
         }
         */
 
-        var json = "{\"rfc3339\":\"2009-11-10T23:00:00Z\",\"rfc3339micro\":\"2009-11-10T23:00:00.000000Z\",\"rfc3339nano\":\"2009-11-10T13:00:00.000000000Z\"}\r\n";
+        // second changed from go playground
+        var json = "{\"rfc3339\":\"2009-11-10T23:45:12Z\",\"rfc3339micro\":\"2009-11-10T23:45:12.123456Z\",\"rfc3339nano\":\"2009-11-10T13:24:56.123456789Z\"}\r\n";
 
```

Comment:
Please consider the following datetimes as test-cases, that are valid with regard to Kubernetes-Api, but all lead to a FormatException, because they not match 3, 6, or 9 digits:

"2009-11-10T13:24:56.12345678Z"
"2009-11-10T13:24:56.12345Z"
"2009-11-10T13:24:56.1234Z"
"2009-11-10T13:24:56.12Z"
"2009-11-10T13:24:56.1Z"

Reply:
any reason to support those format that server will never output?

Reply:
Hello @tg123. As already described in issue https://github.com/kubernetes-client/csharp/issues/1350, timestamps in the Kubernetes API are implemented by the Golang type [`time.Time`](https://pkg.go.dev/time#Time) which implicitly defines what is and is not allowed in terms of nanosecond digits.

Parsing a string as this type allows an arbitrary number of nanosecond digits from 0-9. And in fact, also deserializing an instance of this type may lead to an arbitrary number of nanosecond digits from 0-9. Consider [this example](https://go.dev/play/p/VWdgDKRBWHH).

I currently see nothing that explicitly excludes that this actually happens in practice and see the burden of proof for the contrary more on your side.

Reply:
i do not think it should be supported through
why not if just one line change

Reply:
I don't understand your second line (why not if just one line change). For the first line: I think it doesn't matter what we think about whether arbitrary nanosecond digits should be supported or not - all what matters is that the Kubernetes API permits it. And as long as the C# Kubernetes Client does not support it, it will fail sooner or later in a very unpleasant and subtil way. I think an a bit more complex implementation of the deserialization is a small price for avoiding this.

Reply:
Sorry for the misunderstanding! Your solution is actually very simple and elegant. Thank you :-)